expect.go

```diff
@@ -318,8 +318,19 @@ func WithConfig(config Config) *Expect {
 	if config.WebsocketDialer == nil {
 		config.WebsocketDialer = &websocket.Dialer{}
 	}
+	if config.Formatter == nil {
+		config.Formatter = DefaultFormatter{}
+	}
 	return &Expect{
 		config: config,
+		context: &Context{
+			TestName:  "", // FIXME: should be set here?
```

Comment:
I think we can add TestName to Config. If it's not set, we can check whether Reporter stored in Config supports Name() method. If it does, we can call it and get the name. It's implemented by testing.T.

To deal with the situation when Reporter is not testing.T, but is AssertReporter or RequireReporter, we can add Name() method to them too, which in turn calls Name() on the underlying object (if it has it).

With this approach, in most cases we will determine test name automagically, but the user will still have a possibility to set/override it by hand.